### PR TITLE
Add pre-commit hook

### PR DESCRIPTION
**Pull request by manishsaha**
## Overview
Some big developer changes here. Added a pre-commit hook using `husky` and `lint-staged` to enforce uniformity across contributors and codebase. Using `estlint` and `prettier` for linting and formatting.

## Changes Made
- Since tslint has been deprecated, I added `eslint-config-airbnb-typescript-prettier` (the new kid on the block), combining `eslint` and `prettier` for code style.
- Lint all files following airbnb style.
- Redacted `gts` and updated `npm` run scripts for formatting and linting.
- Added pre-commit hook using `husky` and `lint-staged` to check staged files (i.e. files that will be committed).
- Fixed a _lot_ of npm vulnerabilities.

## Test Coverage
Pre-commit works! Tested using dummy commits and checking whether `husky` is doing its thing.

## Next Steps
Look into adding `hound` for this repo.

### CODE REVIEW

**Review by alanna-zhou — APPROVED** [review]
wau

File: src/routers/GetUsersRouter.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ class GetUsersRouter extends AuthenticatedAppplicationRouter<SerializedUser[]> {
 
   async content(req: Request): Promise<SerializedUser[]> {
     const users = await UserRepo.getUsers();
-    return users.map(userObject => userObject.serialize());
+    return users.map((userObject) => userObject.serialize());
```
Comment (chalo2000) [inline src/routers/GetUsersRouter.ts:17]: I'm assuming the linter wasn't converting interestObject variables into interest as above so you probably missed converting this one to just user

File: src/routers/UpdateDemographicsRouter.ts [hunk 2]
```diff
@@ -16,23 +15,17 @@ class UpdateDemographicsRouter extends AuthenticatedAppplicationRouter<void> {
...
 
     /* sanitize fields */
-    for (const key in body) {
-      if (!validFields.includes(key) || key === 'googleID' || key === 'netID') {
+    Object.keys(body).forEach((key) => {
+      if (!validFields.includes(key)) {
         throw Error(`Invalid field '${key}' identified in request body.`);
       }
-    }
+    });
```
Comment (chalo2000) [inline src/routers/UpdateDemographicsRouter.ts:25]: goodbye having to bother with adding semicolons 🤩

**Review by chalo2000 — APPROVED** [review]
yay

**Pull request merged**